File: packages/js-sdk/src/regex/detectors/it/partita-iva.ts
// Italian Partita IVA (VAT number) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { itPartitaIvaChecksum } from '../../validators'

class ItPartitaIvaDetector extends RegexDetector {
  entityType = EntityType.IT_PARTITA_IVA
  score = 0.85
  contextKeywords = ['partita iva', 'p.iva', 'p. iva', 'vat', 'iva']
  contextRequired = true

  pattern = /\b(?:IT)?\d{11}\b/gi
  validator = itPartitaIvaChecksum
}

registerRegion('it', ItPartitaIvaDetector)

export { ItPartitaIvaDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/it/partita_iva.py
"""Italian Partita IVA (VAT number) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import it_partita_iva_checksum


@register_region("it")
class PartitaIvaDetector(RegexDetector):
    entity_type = EntityType.IT_PARTITA_IVA
    score = 0.85
    context_keywords = [
        "partita iva", "p.iva", "p. iva", "vat", "iva",
    ]
    context_required = True
    context_window = 50

    # Optional IT prefix + 11 digits
    pattern = re.compile(
        r"\b(?:IT)?\d{11}\b",
        re.IGNORECASE,
    )
    validator = staticmethod(it_partita_iva_checksum)
